components/centraldashboard/app/api.ts

```diff
@@ -84,26 +122,73 @@ export class Api {
     return {
       user: user.email,
       platform,
-      namespaces: this.mapNamespacesToWorkgroupBindings(user.email, namespaces),
+      namespaces: this.mapNamespacesToSimpleBinding(user.email, namespaces),
       isClusterAdmin: true,
     };
   }
 
   /**
-   * Converts Kubernetes Namespace types to WorkgroupBindings to ensure
+   * Converts Kubernetes Namespace types to SimpleBinding to ensure
    * compatibility between identity-aware and non-identity aware clusters
    */
-  private mapNamespacesToWorkgroupBindings(
-      user: string, namespaces: V1Namespace[]): WorkgroupBinding[] {
+  private mapNamespacesToSimpleBinding(
+      user: string, namespaces: V1Namespace[]): SimpleBinding[] {
     return namespaces.map((n) => ({
-                            user: {kind: 'user', name: user},
-                            referredNamespace: n.metadata.name,
-                            roleRef: {
-                              apiGroup: '',
-                              kind: 'ClusterRole',
-                              name: 'editor',
-                            },
-                          }));
+      user,
+      namespace: n.metadata.name,
+      role: roleMap.tr('edit'),
+    }));
+  }
+  /**
+   * Converts Workgroup Binding from Profile Controller to SimpleBinding
+   */
+  private mapWorkgroupBindingToSimpleBinding(bindings: WorkgroupBinding[]): SimpleBinding[] {
+    return bindings.map((n) => ({
+      user: n.user.name,
+      namespace: n.referredNamespace,
+      role: roleMap.tr(n.roleRef.name),
+    }));
+  }
+  /**
+   * Converts SimpleBinding to Workgroup Binding from Profile Controller
+   */
+  private mapSimpleBindingToWorkgroupBinding(binding: SimpleBinding): WorkgroupBinding {
+    const {user, namespace, role} = binding;
+    return {
+      user: {
+        kind: 'User',
+        name: user,
+      },
+      referredNamespace: namespace,
+      roleRef: {
+        kind: 'ClusterRole',
+        name: roleMap.reverse(role),
+      }
+    };
+  }
+
+  /**
+   * Handles an exception in an async block and converts it to a JSON
+   * response sent back to client
+   */
+  // tslint:disable-next-line: no-any
+  private surfaceProfileControllerErrors = (info: {res: Response, msg: string, err: any}) => {
+    const {res, msg, err} = info;
+    const code = (err.response && err.response.statusCode) || 400;
+    const error = err.body || msg;
+    console.log(`${msg} ${error}`);
+    apiError({res, code, error});
+  }
+
+  async getContributors(namespace: string) {
+    const {body} = await this.profilesService
+      .readBindings(undefined, namespace);
+    const users = this.mapWorkgroupBindingToSimpleBinding(body.bindings)
+      .filter((b) => b.role === 'contributor')
+      .map((b) =>
+        b.user
```

Comment:
Can this be moved up to be on the same line as `.map()` call?

Reply:
Done